src/Take.Blip.Builder/Actions/ExecuteScript/ExecuteScriptAction.cs

```diff
@@ -59,5 +78,17 @@ public override async Task ExecuteAsync(IContext context, ExecuteScriptSettings
                 await context.DeleteVariableAsync(settings.OutputVariable, cancellationToken);
             }
         }
+
+        private StepMode CheckMemoryUsage(IContext context, DebugInformation debugInformation)
+        {
+            if (debugInformation.CurrentMemoryUsage >= _configuration.ExecuteScriptLimitMemoryWarning)
+            {
+                var currentActionTrace = context.GetCurrentActionTrace();
+                currentActionTrace.Warning = $"Script is using so much memory allocation({debugInformation.CurrentMemoryUsage})." +
```

Comment:
This can be null. You should set the message only when there is an active trace.

```suggestion
if (currentActionTrace != null)
{
                currentActionTrace.Warning = $"Script is using so much memory allocation({debugInformation.CurrentMemoryUsage})." +
```

Reply:
Change the message for something like:
```csharp
$"The script memory usage is above the warning threshold of {_configuration.ExecuteScriptLimitMemoryWarning} bytes"
```